packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -977,6 +988,42 @@ export class MergeTree {
             } else {
                 const segment = node;
                 const removalInfo = toRemovalInfo(segment);
+                if (this.options?.mergeTreeUseNewLengthCalculations === true) {
+                    // normalize the seq numbers
+                    // if the remove is local (UnassignedSequenceNumber) give it the highest possible
+                    // seq for comparison, as it will get a seq higher than any other seq once sequenced
+                    // if the segments seq is local (UnassignedSequenceNumber) give it the second highest
+                    // possible seq, as the highest is reserved for the remove.
+                    const seq = node.seq === UnassignedSequenceNumber
+                        ? Number.MAX_SAFE_INTEGER - 1
+                        : node.seq ?? 0;
+
+                    if (removalInfo !== undefined) {
+                        const removedSeq = removalInfo.removedSeq === UnassignedSequenceNumber
+                            ? Number.MAX_SAFE_INTEGER
+                            : removalInfo?.removedSeq;
+                        if (removedSeq <= this.collabWindow.minSeq) {
+                            return undefined;
+                        }
+                        if (removedSeq <= refSeq || removalInfo.removedClientIds.includes(clientId)) {
+                            return 0;
+                        }
+                    }
+
+                    if (seq <= refSeq || segment.clientId === clientId) {
+                        return segment.cachedLength;
+                    } else {
+                        // the segment was inserted and removed before the
```

Comment:
clauses here imply `seq > refSeq && segment.clientId !== clientId && removalInfo !== undefined && removalInfo.removedSeq !== UnassignedSequenceNumber`, which looks more like the segment was inserted and removed *after* the context. But it must be within the collab window since `seq > refSeq`. Seems confusing.

Reply:
i think you are right. i didn't actually change that part in the original pr, it's the same below, but it shows up now that i fork the paths base on the config.  the right thing to do is just return zero, as the client didn't see it, but it should still participate until out of the collab window. nice catch!

Reply:
yeah, I remember this comment confusing me before the changes. But I recently re-read the DEV.md description of the undefined vs. 0 distinction and got some more confidence that it wasn't just my lack of understanding :)

Reply:
actually, i did the same thing in the other PR, just missed it in the copy paste :(